core/shared/src/main/scala/zio/stm/STM.scala

```diff
@@ -537,10 +547,9 @@ object STM {
           value match {
             case _: TRez.Succeed[_] =>
               if (analysis eq JournalAnalysis.ReadWrite) {
-                globalLock.acquire()
-
-                try if (isValid(journal)) commitJournal(journal) else loop = true
-                finally globalLock.release()
+                globalLock.synchronized {
```

Comment:
I thought `.synchronized` threw exceptions on Scala.js. No???

Reply:
Yes, java built-in `synchronized` throws exception on Scala.js.
So, I implemented and used `synchronized` method on `LockedRef` class which wraps the procedure in `{}` code with locking/unlocking.

Reply:
@heraklos How about this:

```scala
// JS
object Sync {
  def apply[A](anyRef: AnyRef)(f: => A): A = f
// JVM
object Sync {
  def apply[A](anyRef: AnyRef)(f: => A): A = anyRef.synchronized { f }
}
```

Then you can replace `x.synchronized { f }` with `Sync(x) { f }`

Sound good?

Reply:
@jdegoes 
That sounds perfect to me!
And with `Sync`, `STMSpec` passed for both `jvm`/`js`!

The only obstacle here is the following error 
```
parameter value anyRef in method apply is never used
```
for `Sync` of `js`.
```
// JS
object Sync {
  def apply[A](anyRef: AnyRef)(f: => A): A = f
}
```
I'm gonna think of some workaround.